### PR TITLE
Clarify whether core packages are required to make bug-fix releases

### PR DESCRIPTION
**Pull request by jarrodmillman**
@thomasjpfan This is a first draft to address the confusion / concerns raised at a recent scikit-learn developer meeting (https://discord.com/channels/786703927705862175/1201956709418139778).  The SPEC committee is meeting (https://scientific-python.org/calendars/) tomorrow (2/6) and we will discuss this attempt to clarify the SPEC intent. If you have any additional thoughts / suggestions, feel free to comment here (you are also more than welcome to join the meeting).

### CODE REVIEW

File: spec-0000/index.md [hunk 1]
```diff
@@ -24,7 +24,8 @@ endorsed-by:
 
 ## Description
 
-This SPEC recommends that all projects across the Scientific Python ecosystem adopt a common time-based policy for dropping support of older Python and core package versions.
+This SPEC recommends that all projects across the Scientific Python ecosystem adopt a common time-based policy for dropping support
+for older Python and for support for older versions of core packages depended upon by those projects.
```
Comment (stefanv) [inline spec-0000/index.md:28]:
```suggestion
This SPEC recommends that all projects across the Scientific Python ecosystem adopt a common time-based policy for dropping support for core packages depended upon by those projects, as well as older versions of Python.
```

File: spec-0000/index.md [hunk 2]
```diff
@@ -33,12 +34,26 @@ Specifically, we recommend that:
...
 ### Core Project Endorsement
 
 <!--
 Briefly discuss what it means for a core project to endorse this SPEC.
 -->
 
+Core project endorsing this SPEC means that those projects encourage all projects across the Scientific Python ecosystem
+to limit how long they support older Python versions and older core packages versions.
+While core projects may wish to provide bug-fix releases of older versions, endorsing this SPEC does **not** imply that
+core projects will provide bug-fix releases during the full 2 year period after release.
```
Comment (stefanv) [inline spec-0000/index.md:53]:
```suggestion
A core project endorsing this SPEC does **not** imply that that project will provide bug-fix releases for two full years after a release.
```

File: spec-0000/index.md [hunk 3]
```diff
@@ -33,12 +34,26 @@ Specifically, we recommend that:
 1. Support for Python versions be dropped **3 years** after their initial release.
 2. Support for core package dependencies be dropped **2 years** after their initial release.
 
+{{< admonition note >}}
+Core packages may or may not to provide bug fix releases during the full 2 year period after release.
+So projects may decide to occassionally drop support for core pakage dependencies earlier than recommended by this SPEC.
```
Comment (stefanv) [inline spec-0000/index.md:39]:
```suggestion
Therefore, projects may occassionally want to drop support for core package dependencies earlier than recommended by this SPEC.
```

File: spec-0000/index.md [hunk 4]
```diff
@@ -33,12 +34,26 @@ Specifically, we recommend that:
...
 <!--
 Briefly discuss what it means for a core project to endorse this SPEC.
 -->
 
+Core project endorsing this SPEC means that those projects encourage all projects across the Scientific Python ecosystem
+to limit how long they support older Python versions and older core packages versions.
+While core projects may wish to provide bug-fix releases of older versions, endorsing this SPEC does **not** imply that
+core projects will provide bug-fix releases during the full 2 year period after release.
+As many of the core projects are volunteer projects with limited developer time, core package developers may often prioritize
+new features and enhancements over long-term package maintenance.
```
Comment (tupui) [inline spec-0000/index.md:53]: I don't think we should add that note. For some projects that could be argued and here that would not really help the messaging IMHO. At most I would say something along the line of "project's developments are discussed and prioritised with and by the community".

**Comment by jarrodmillman** [comment]
This may be a good time to start fleshing out the `Core Project Endorsement` and `Ecosystem Adoption` sections.

File: spec-0000/index.md (hunk 4 above)
Comment (bsipocz) [reply spec-0000/index.md:53]: I agree that it's better without the last sentence.

File: spec-0000/index.md [hunk 5]
```diff
@@ -24,7 +24,7 @@ endorsed-by:
 
 ## Description
 
-This SPEC recommends that all projects across the Scientific Python ecosystem adopt a common time-based policy for dropping support of older Python and core package versions.
+This SPEC recommends that all projects across the Scientific Python ecosystem adopt a common time-based policy for dropping support for core packages depended upon by those projects, as well as older versions of Python.
```
Comment (stefanv) [inline spec-0000/index.md:27]:
```suggestion
This SPEC recommends that all projects across the Scientific Python ecosystem adopt a common time-based policy for dropping dependencies. From the perspective of this SPEC, the dependencies in question are core packages as well as older Python versions.
```

File: spec-0000/index.md [hunk 6]
```diff
@@ -33,12 +33,25 @@ Specifically, we recommend that:
 1. Support for Python versions be dropped **3 years** after their initial release.
 2. Support for core package dependencies be dropped **2 years** after their initial release.
 
+{{< admonition note >}}
+Core packages may or may not to provide bug fix releases during the full 2 year period after release.
```
Comment (Carreau) [inline spec-0000/index.md:37]:
```suggestion
Core packages may or may not decide to provide bug fix releases during the full 2 year period after release.
```

missing verb ?

File: spec-0000/index.md [hunk 7]
```diff
@@ -33,12 +33,25 @@ Specifically, we recommend that:
...
 
 <!--
 Briefly discuss what it means for a core project to endorse this SPEC.
 -->
 
+Core project endorsing this SPEC means that those projects encourage all projects across the Scientific Python ecosystem
+to limit how long they support older Python versions and older core packages versions.
+A core project endorsing this SPEC does **not** imply that that project will provide bug-fix releases for two full years after a release.
+As many of the core projects are volunteer projects with limited developer time, core package developers may often prioritize
+new features and enhancements over long-term package maintenance.
```
Comment (jarrodmillman) [inline spec-0000/index.md:53]:
```suggestion
```

File: spec-0000/index.md [hunk 8]
```diff
@@ -33,12 +33,23 @@ Specifically, we recommend that:
...
+{{< /admonition >}}
+
 ### Core Project Endorsement
 
 <!--
 Briefly discuss what it means for a core project to endorse this SPEC.
 -->
 
+Core project endorsing this SPEC means that those projects encourage all projects across the Scientific Python ecosystem
+to limit how long they support older Python versions and older core packages versions.
```
Comment (bsipocz) [inline spec-0000/index.md:50]: This SPEC only has a table for the core projects, but the driving philosophy is true for all dependencies I believe.

```suggestion
to limit how long they support older Python versions and older dependency versions.
```

**Review by bsipocz — APPROVED** [review]
Thank you for the clarification. One minor comment, take it or leave it.

**Review by stefanv — APPROVED** [review]
I think we could generalize this even further to non-core projects, but will propose that in a follow-up PR, in case it generates further discussion.

**Comment by bsipocz** [comment]
As people are generally happy with the changes and it was also discussed in today's SPEC meeting, too, I am going ahead and merging this now.

Thanks!

**Pull request merged**